tests/test_visitors/test_ast/test_naming/test_variable_self_reassignment.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from wemake_python_styleguide.violations.best_practices import (
+    ReassigningVariableToItselfViolation,
+)
+from wemake_python_styleguide.visitors.ast.naming import (
+    WrongVariableAssignmentVisitor,
+)
+
+wrong_fragment = """
+test_variable = 5
+test_variable = test_variable
+"""
+
+right_fragment = """
+test_variable = 5
+test_variable = 10
+"""
+
+wrong_fragment_double_assignment = """
+test_variable = 5
+test_variable = test_variable = 10
+"""
+
+wrong_fragment_other_assignment = """
+test_variable = 5
+test_variable = other = test_variable = 5
+"""
+
+wrong_fragment_tuple_assignment = """
+x = 1
+y = 2
+x, y = x, y
```

Comment:
One more thing. Can you please test this case:

```python
x = 1
y = 2
x, y = y, x
```

This should be perfectly valid case. But, I am not sure if that is working right now.